controllers/mfa.go

```diff
@@ -104,32 +107,38 @@ func (c *ApiController) MfaSetupVerify() {
 		MfaType: mfaType,
 	}
 	if mfaType == object.TotpType {
-		secret := c.GetSession(MfaTotpSecretSession)
-		if secret == nil {
-			c.ResponseError("totp secret is missing")
-			return
-		}
-		config.Secret = secret.(string)
-	} else if mfaType == object.SmsType {
-		dest := c.GetSession(MfaDestSession)
-		if dest == nil {
-			c.ResponseError("destination is missing")
-			return
+		secret := c.Ctx.Request.Form.Get("secret")
```

Comment:
define

```go
c.getSessionOrFormValue(field string) (string, error) {
res:= c.Ctx.Request.Form.Get(field)
if res != nil {
  return res, nil
}

res, ok := c.GetSession(fmt.Sprintf("mfa_%s", field)).(string)
if ok {
  return res, nil
}

return "", fmt.Errorf("MFA field: %s is missing", field)
}
```

Similar changes for all other ocurrances

Similar changes for SetSession, DelSession, etc.

Reply:
I refactored the code. Regarding SetSession and DelSession, I need to determine if the request is indeed stateless, see isSessionOidc.